include/zephyr/net/websocket.h

```diff
@@ -166,7 +166,10 @@ int websocket_send_msg(int ws_sock, const uint8_t *payload, size_t payload_len,
  *        The value is in milliseconds. Value SYS_FOREVER_MS means to wait
  *        forever.
  *
- * @return <0 if error, >=0 amount of bytes received
+ * @retval >=0 amount of bytes received.
+ * @retval -EAGAIN on timeout.
+ * @retval -ENOTCONN on socket close.
```

Comment:
As we update the API, we should update the code that relied on the old, 0 on disconnect behaviour. I've identified two places:
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/lib/mqtt/mqtt_transport_websocket.c - should return 0 on `-ENOTCONN`
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/lib/websocket/websocket.c#L1027 - same story

Reply:
ALso i found

https://github.com/zephyrproject-rtos/zephyr/blob/8bc094610c5dd69f6ba0085fb3fe7d18167164c9/samples/net/sockets/websocket_client/src/main.c#L204